browser/ui/views/infobars/brave_confirm_infobar.cc

```diff
@@ -25,14 +34,83 @@ std::unique_ptr<infobars::InfoBar> CreateBraveConfirmInfoBar(
 
 BraveConfirmInfoBar::BraveConfirmInfoBar(
     std::unique_ptr<BraveConfirmInfoBarDelegate> delegate)
-    : ConfirmInfoBar(std::move(delegate)) {
-  auto* delegate_ptr = GetBraveDelegate();
-  DCHECK(delegate_ptr);
+    : InfoBarView(std::move(delegate)) {
+  auto* delegate_ptr = GetDelegate();
+  label_ = AddChildView(CreateLabel(delegate_ptr->GetMessageText()));
+  label_->SetElideBehavior(delegate_ptr->GetMessageElideBehavior());
+
+  const auto create_button =
+      [this](ConfirmInfoBarDelegate::InfoBarButton type,
+             void (BraveConfirmInfoBar::*click_function)()) {
+        auto* button = AddChildView(std::make_unique<views::MdTextButton>(
+            base::BindRepeating(click_function, base::Unretained(this)),
```

Comment:
This was copied from ConfirmInfobar, i will use weak ptr here

Reply:
I think we don't need weak ptr here because button is owned by `BraveConfirmInforBar`.
Passed |click_function| is valid always till |button| is gone.